app/Controllers/customerController.php

```diff
@@ -0,0 +1,127 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * This controller handles action about authentication.
+ */
+class FreshRSS_customer_Controller extends FreshRSS_ActionController {
+	public function payAction(): void {
+		if (FreshRSS_Auth::hasAccess()) {
+			Minz_Request::forward(['c' => 'index', 'a' => 'index'], true);
+		}
+
+		if (max_registrations_reached()) {
+			Minz_Error::error(403);
+		}
+		require '../../vendor/autoload.php';
+		\Stripe\Stripe::setAppInfo(
+			"news.decent.im",
+			"0.0.1",
+			"https://github.com/decent-im/FreshRSS"
+		);
+		\Stripe\Stripe::setApiKey(FreshRSS_Context::systemConf()->stripe_secret_key);
+
+		// Create new Checkout Session for the order
+		// Other optional params include:
+		// [billing_address_collection] - to display billing address details on the page
+		// [customer] - if you have an existing Stripe Customer ID
+		// [payment_intent_data] - lets capture the payment later
+		// [customer_email] - lets you prefill the email input in the form
+		// [automatic_tax] - to automatically calculate sales tax, VAT and GST in the checkout page
+		// For full details see https://stripe.com/docs/api/checkout/sessions/create
+
+		// ?session_id={CHECKOUT_SESSION_ID} means the redirect will have the session ID set as a query param
+		$success_url = Minz_Url::display(
+			['c' => 'customer', 'a' => 'issueTicket'],
+			/*encoding*/'raw', /*absolute=*/true);
+		$cancel_url = Minz_Url::display(
+			['c' => 'customer', 'a' => 'payFailed'],
+			/*encoding*/'raw', /*absolute=*/true);
+		$checkout_session = \Stripe\Checkout\Session::create([
+			'success_url' => $success_url . '&session_id={CHECKOUT_SESSION_ID}',
+			'cancel_url' => $cancel_url,
+			'mode' => 'subscription',
+			// 'ui_mode' => 'embedded', // TODO for another day
+			'line_items' => [[
+				'price' => FreshRSS_Context::systemConf()->stripe_price_id,
+				'quantity' => 1,
+			]],
+			'subscription_data' => [
+				'trial_period_days' => 30,
+			],
+		]);
+
+		header($_SERVER["SERVER_PROTOCOL"] . " 303 Proceed to payment page");
+		header("Location: " . $checkout_session->url);
+		exit;
+	}
+
+	public function issueTicketAction(): void {
+		require '../../vendor/autoload.php';
```

Comment:
suggestion : Hi, Shouldn't the call for autoload be made in such a way as to benefit the whole project